MainEvent/main.cpp

```diff
@@ -94,6 +101,51 @@ void MoveRobot() {
   //Roboteq.sendAngles();
 }
 
+bool sortingMethod (Mat& a, Mat& b){
+    return a.Y > b.Y;
+}
+
+void matchingMethod(){
+  // DONE: Write function here to match the balls in the ballCoordL & ballCoordR and put them into vector of pair
+  //sorting here is inefficient(not like it really matters that much here), makes more sense to sort after balls out of bounds removed
+  //sort(ballCoordR.begin(), ballCoordR.end(), sortingMethod());
+  //sort(ballCoordL.begin(), ballCoordL.end(), sortingMethod());
+
+  vector<Mat> correctedR;
+  vector<Mat> correctedL;
+
+  for (auto lball: ballCoordL){
+    if (lball.X > leftBound){
+      correctedL.push_back(lball);
+    }
+  }
+  for (auto rball: ballCoordR){
+    if (rball.X < rightBound){
+      correctedR.push_back(rball);
+    }
+  }
+
+  sort(correctedR.begin(), correctedR.end(), sortingMethod());
+  sort(correctedL.begin(), correctedL.end(), sortingMethod());
+  //balls are now sorted by their Y values
+  //how much does sorting even do here?
+
+  for (auto rball: correctedR){
+    for (auto lball: correctedL){
+      if (rball.Y == lball.Y){
```

Comment:
The Y positions won't be exactly equal, should be a range
```suggestion
      if (abs(rball.Y - lball.Y) < 10){
```
10 pixels was random, would have to find what it actually should be

Reply:
turned 10 into marginOfError to simplify things